cdk-apigw-authorizer-sqs/README.md

```diff
@@ -0,0 +1,97 @@
+
+![Alt text](images/apigw-arch.png?raw=true "Pattern using API gateway, SQS and Lambda Authorizer")
+
+
+This CDK project is written in Python, it has 2 stacks and can synthesize 2 cloud formation templates. Customers can deploy either of the stacks based on their API authorization needs:
+ 
+##### Stack 'ApigwSqsAuthStack' will create:
+* API Gateway Rest API with a POST method acting as a proxy to SQS,
+* A Token Authorizer for the API Gateway ( Lambda ARN of the function implementing the authorization is provided as a input parameter to CF and CDK)
+* Amazon SQS queue to store the payload posted
+
+##### Stack 'APigwSqsStack' will create:
+* API Gateway Rest API with a POST method acting as a proxy to SQS,
+* Amazon SQS queue to store the payload posted
+
+
+Important: this application uses various AWS services and there are costs associated with these services after the Free Tier usage - please see the [AWS Pricing page](https://aws.amazon.com/pricing/) for details. You are responsible for any AWS costs incurred. No warranty is implied in this example.
+
+## Requirements
+
+* [Create an AWS account](https://portal.aws.amazon.com/gp/aws/developer/registration/index.html) if you do not already have one and log in. The IAM user that you use must have sufficient permissions to make necessary AWS service calls and manage AWS resources.
+* [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html) installed and configured
+* [Git Installed](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+* [CDK Installed](https://docs.aws.amazon.com/cdk/v2/guide/getting_started.html) 
+* [Create Lambda authorizer from blueprint](https://github.com/awslabs/aws-apigateway-lambda-authorizer-blueprints/blob/master/blueprints/python/api-gateway-authorizer-python.py)
+
+![Alt text](images/lambda-auth-blueprint.png?raw=true "Create lambda function using blueprint")
+
+
+## Deployment Instructions
+
+1. Create a new directory, navigate to that directory in a terminal and clone the GitHub repository:
+```
+git clone https://github.com/aws-samples/pattern-apigw-auth-sqs.git
+```
+2. Change directory
+```
+cd pattern-apigw-auth-sqs
+```
+3. To generate a cloudformation templates (optional)
+* With Token Autorizer
+    ```
+    cdk synth ApigwSqsAuthStack -o ./cloudformation_templates
+    ```
+* Without a Token Autorizer
+    ```
+    cdk synth ApigwSqsStack -o ./cloudformation_templates
+    ```
+4. To deploy AWS resources as a CDK project
+   ##### Paramters needed for CDK 
+    * storagequeue=Name of the SQS queue where you want data to be sent
```

Comment:
what queue? is not in the prerequistes. Can you create it as Iac?

Reply:
The SQS queue will be created by IaC. This parameter will take a value for 'Queue Name' - Users can provide a user friendly name that will be created by IaC.